### ISSUE TITLE
Allow sortable headers with generic choosers

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#11794]
### Is your proposal related to a problem?

It is quite easy to setup and customize the chooser modal. For example, you can add columns easily.
Unfortunately, you cannot use the "sort_key" to enable sortable columns, since those sort-headers aren't handled by the modal's JS.

That's probably the reason, why the `sort_key` is set to `None` here:
https://github.com/wagtail/wagtail/blob/79bc23788b31337012a43533f4879e7361a02b7f/wagtail/snippets/views/chooser.py#L33

### Describe the solution you'd like

It would be nice to have this handled by wagtail's JS. I think this would probably also help wagtailmedia's outdated [implementation](https://github.com/torchbox/wagtailmedia/blob/327267e0da6c81413ed4a521ebf8e4ad76cc395e/src/wagtailmedia/static/wagtailmedia/js/media-chooser-modal.js) (which allows sorting) to be ported more easily.

There are similar tickets for the page chooser, which could also benefit from that functionality:
https://github.com/wagtail/wagtail/issues/3544
https://github.com/wagtail/wagtail/issues/8557

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by Srvankit** [comment]
Hi! I’d like to work on this issue.

From my understanding, the problem is that while chooser columns can define a sort_key, the chooser modal UI does not currently handle sorting interactions, so users cannot click headers to sort results.

My proposed approach:
- Investigate how sorting is handled in other chooser implementations (e.g., page chooser)
- Extend the chooser modal JS to support clickable sortable headers
- Ensure sort_key is respected and passed correctly to the backend

Does this approach align with what you're expecting? Happy to refine it before starting implementation.
